Added "Why Senza" section to readme

rafaelcaricio opened the pull request:
This PR is part of an emerging effort to make Senza more newbie friendly.

LappleApple commented:
@rafaelcaricio @jmcs Once you merge this small PR I'll take a look and make any small fixes needed

rafaelcaricio commented:
@LappleApple Could you please review it and fix things in the PR, so we merge only when it is ready and with good formatting? I have added you to my fork so you can commit directly.

LappleApple commented:
done

hjacobs commented on README.rst:
```diff
@@ -102,9 +105,9 @@ Senza Definition
-See the `STUPS documentation on Senza`_ for details.
+See the `STUPS.io documentation on Senza`_ for details.
 
-.. _STUPS documentation on Senza: http://stups.readthedocs.org/en/latest/components/senza.html
```
we should also change readthedocs.org to readthedocs.io

hjacobs commented:
@rafaelcaricio can you update JSON -> YAML so we can merge?

rafaelcaricio commented:
@hjacobs I am reviewing the text. Some parts are wrong/make no sense after the review -- I am trying to fix it now.

LappleApple commented:
@rafaelcaricio @hjacobs Can you point out what makes sense? I mostly rearranged the sentences in order and deleted a redundancy. One original sentence--"Although writing CloudFormation in JSON format is not human friendly and many parts of a CloudFormation template are reusable among applications of the same kind."--wasn't actually a sentence, so I rewrote it. 

Also check to make sure it's always "CloudFormation" and not "Cloud Formation."

rafaelcaricio commented:
@LappleApple I think this misunderstanding was because my text wasn't clear enough. But what I fixed was this part 

> However, writing CloudFormation templates in JSON format is not human-friendly, which hinders developer productivity and consistency among deployed stacks.

The fact of writing CloudFormation in JSON does not have anything to do with the consistency among deployed stacks. I just removed this part in my new review. Please check the latest commit.

LappleApple commented:
Other than adding a comma after your new "Also" (I didn't), this part is still confusing:

_"Also, many parts of a CloudFormation template are reusable among applications of the same kind. Senza addresses this problem by supporting CloudFormation templates as YAML input and adding its own 'components' on top."_

The way it's written suggests that the reusability of different parts of the CF template is a problem ("this problem"). Is this characteristic of CF templates actually a problem?

rafaelcaricio commented:
No, the problem is not being able to reuse. Please check the text to see if I messed up in something more.

hjacobs commented:
:+1:
